compose/project.py

```diff
@@ -473,6 +477,23 @@ def get_networks(service_dict, network_definitions):
     return networks
 
 
+def match_named_volumes(service_dict, project_volumes):
+    for volume_spec in service_dict.get('volumes', []):
+        if volume_spec.is_named_volume:
+            declared_volume = next(
+                (v for v in project_volumes if v.name == volume_spec.external),
+                None
+            )
```

Comment:
Instead of doing a n^2 lookup, creating a dict keyed by external name would make this an n.